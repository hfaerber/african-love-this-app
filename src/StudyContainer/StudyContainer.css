.div-map-image {
  border: 4px solid purple;
  width: 100%;
  text-align: left;
  display: flex;
}

.img-map {
  max-width: 50%;
  border: 4px solid red;
  max-height: 59vh;
}

.section-card-display {
  display: flex;
  flex-wrap: wrap;
  width: 60%;
  border: 2px solid blue;
  overflow: scroll;
}
