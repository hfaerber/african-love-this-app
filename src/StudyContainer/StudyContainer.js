import React from 'react';
import './StudyContainer.css';
// import CountryCard from '../CountryCard';
import colormap from '../images/africa-map-countries-capitals-color.jpg'


const StudyContainer = () => {
  return (
    <div className='div-map-image'>
      <img src={colormap} className="img-map"
        alt="map of africa with capital cities" />

    </div>
  )
}

export default StudyContainer;

// display map and give button click fn to enlarge
// create and display <country cards/>
// display <filter and search container> (stateful?)
