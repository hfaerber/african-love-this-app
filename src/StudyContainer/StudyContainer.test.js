import React from 'react';
import { StudyContainer, mapStateToProps } from './StudyContainer';
import { shallow } from 'enzyme';

describe('StudyContainer', () => {
  describe('StudyContainer component', () => {
    let wrapper, mockCountries;

    beforeEach(() => {
      mockCountries = [
        {name: "Algeria", capital: "Algiers", subregion: "Northern Africa", needsStudied: true},
        {name: "Benin", capital: "Porto-Novo", subregion: "Western Africa", needsStudied: true},
      ];
      wrapper = shallow(<StudyContainer
        countries={mockCountries}
        selectedFilter='All'
        searchQuery=''
        />)
    })

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot;
    })

// it('should invoke filter() if searchQuery is empty string')
  });

  describe('mapStateToProps', () => {
    const mockCountries = [
      {name: "Algeria", capital: "Algiers", subregion: "Northern Africa", needsStudied: true},
      {name: "Benin", capital: "Porto-Novo", subregion: "Western Africa", needsStudied: true},
    ];

    it('should return an object with the correct props', () => {
      const mockState = {
        user: 'Ben',
        countries: mockCountries,
        selectedFilter: 'All',
        searchQuery: ''
      };
      const expected = {
        countries: mockCountries,
        selectedFilter: 'All',
        searchQuery: ''
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    })
  });
});
