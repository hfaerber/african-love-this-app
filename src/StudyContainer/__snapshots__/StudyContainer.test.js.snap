// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudyContainer StudyContainer component should match snapshot when isLoading false and error true 1`] = `
<div
  className="div-map-image"
>
  <Connect(StudyForm) />
  <section
    className="section-card-display"
  >
    <div
      className="div-error"
    >
      <img
        alt="error icon"
        className="error-icon"
        src="error.svg"
      />
      <h3
        className="error-message"
      >
        There is an error
         
      </h3>
    </div>
  </section>
  <img
    alt="map of africa with capital cities"
    className="img-map"
    src="africa-map-countries-capitals-color.jpg"
  />
</div>
`;

exports[`StudyContainer StudyContainer component should match snapshot when isLoading is true 1`] = `
<div
  className="div-map-image"
>
  <Connect(StudyForm) />
  <section
    className="section-card-display"
  >
    <div
      className="div-loader"
    >
      <h3>
        Loading Study Cards...
      </h3>
      <div
        className="div-giphy"
      >
        <iframe
          allowFullScreen={true}
          className="giphy-embed"
          src="https://giphy.com/embed/tkJsL5AIIsg7K"
        />
      </div>
    </div>
  </section>
  <img
    alt="map of africa with capital cities"
    className="img-map"
    src="africa-map-countries-capitals-color.jpg"
  />
</div>
`;

exports[`StudyContainer StudyContainer component should match snapshot when not loading and no error 1`] = `
<div
  className="div-map-image"
>
  <Connect(StudyForm) />
  <section
    className="section-card-display"
  >
    <Connect(CountryCard)
      capital="Algiers"
      key="Algeria"
      name="Algeria"
      needsStudied={true}
      region="Northern Africa"
    />
    <Connect(CountryCard)
      capital="Porto-Novo"
      key="Benin"
      name="Benin"
      needsStudied={false}
      region="Western Africa"
    />
  </section>
  <img
    alt="map of africa with capital cities"
    className="img-map"
    src="africa-map-countries-capitals-color.jpg"
  />
</div>
`;
